ocaml/idl/datamodel_vm.ml

```diff
@@ -1726,6 +1726,42 @@ let set_uefi_mode =
     ~result:(String, "Result from the varstore-sb-state call")
     ~doc:"Set the UEFI mode of a VM" ~allowed_roles:_R_POOL_ADMIN ()
 
+let vm_secureboot_readiness =
+  Enum
+    ( "vm_secureboot_readiness"
+    , [
+        ("not_supported", "VM's firmware is not UEFI")
+      ; ("disabled", "Secureboot is disabled on this VM")
+      ; ( "first_boot"
+        , "Secured boot is enabled on this VM and its NVRAM.EFI-variables is \
+           empty"
+        )
+      ; ( "ready"
+        , "Secured boot is enabled on this VM and PK, KEK, db and dbx are \
+           defined in its EFI variables"
+        )
+      ; ( "ready_no_dbx"
+        , "Secured boot is enabled on this VM and PK, KEK, db but not dbx are \
+           defined in its EFI variables"
+        )
+      ; ( "setup_mode"
+        , "Secured boot is enabled on this VM and PK is not defined in its EFI \
```

Comment:
In which use cases does this state interact with set_uefi_mode call? And how? The latter doesn't affect the database value directly, while this is db-only

Reply:
`set_uefi_mode` calls `varstore-sb-state` which will affect the VM's NVRAM's EFI-variables.
This method only deduct the state of the VM EFI variables.